src/BloomTests/Book/AppearanceSettingsTests.cs

```diff
@@ -289,22 +289,108 @@ public void ToCss_ContainsSettingsFromJson()
             Assert.That(fromSettings, Does.Contain("--cover-languageName-show: none;"));
         }
 
-        [Test]
-        public void ToCss_EmitsSpecialPageMargin_property()
+        [TestCase(@"""pageNumber-show"": false", false)]
+        [TestCase(@"""pageNumber -show"": true", true)]
+        [TestCase(@"""pageNumber-position"": ""automatic""", true)]
+        [TestCase(@"""pageNumber-position"": ""left""", true)]
+        [TestCase(@"""pageNumber-position"": ""center""", true)]
+        [TestCase(@"""pageNumber-position"": ""right""", true)]
+        [TestCase(@"""pageNumber-position"": ""hidden""", false)]
+        public void ToCss_EmitsSpecialPageMargin_property_andHasCorrectPageNumberShow(
+            string settingsFragment,
+            bool multiplicand1Expected
+        )
         {
             var settings = new AppearanceSettings();
             settings.UpdateFromJson(
                 @"
 {
   ""cssThemeName"": ""default"",
-  ""pageNumber-show"": false
+  "
+                    + settingsFragment
+                    + @"
 }"
             );
             var css = settings.ToCss();
-            Assert.That(css, Does.Contain("--pageNumber-show-multiplicand: 0;"));
-            settings.UpdateFromJson("{\"pageNumber-show\": true}");
-            css = settings.ToCss();
-            Assert.That(css, Does.Contain("--pageNumber-show-multiplicand: 1;"));
+            if (multiplicand1Expected)
+            {
+                Assert.That(css, Does.Not.Contain("--pageNumber-show: none;"));
+                Assert.That(css, Does.Contain("--pageNumber-show-multiplicand: 1;"));
+            }
+            else
+            {
+                Assert.That(css, Does.Contain("--pageNumber-show: none;"));
+                // settings already has --pageNumber-show-multiplicand: 0; by default. Check that it's not being overridden
+                Assert.That(css, Does.Not.Contain("--pageNumber-show-multiplicand: 1;"));
+            }
+        }
+
+        [TestCase(@"""pageNumber-position"": ""automatic""", "unset", "unset")]
+        [TestCase(
+            @"""pageNumber-position"": ""left""",
+            "var(--page-margin-left)",
+            "deliberately-invalid"
+        )]
+        [TestCase(@"""pageNumber-position"": ""center""", "50%", "50%")]
+        [TestCase(
+            @"""pageNumber-position"": ""right""",
+            "deliberately-invalid",
+            "var(--page-margin-right)"
+        )]
+        [TestCase(@"""pageNumber-position"": ""hidden""", "unset", "unset")]
+        // [TestCase(null, null, null)]
```

Comment:
Are we going to migrate all books that have `pageNumber-show: false` to `pageNumber-position: hidden`? If not we need to make sure to preserve the hiding somehow...